src/sentry/middleware/integrations/parsers/msteams.py

```diff
@@ -46,17 +53,29 @@ def get_integration_from_request(self) -> Integration | None:
             return Integration.objects.filter(id=integration.id).first()
         return None
 
+    @classmethod
+    def _check_if_event_should_be_sync(cls, data: Mapping[str, Any]) -> bool:
+        """
+        Determine if an event should be handled synchronously, or if we can defer to async.
+        """
+        raw_event_type = data.get("type", None)
+        if raw_event_type is None:
+            return True
+
+        event_type = MsTeamsEvents.get_from_value(value=raw_event_type)
+        return event_type in cls._synchronous_events
```

Comment:
just a question - why is installation the only event required to be handled synchronously?

Reply:
This is the event we get during an installation for teams, and we want to unblock them first so they can properly install and setup issue alerts